### ISSUE TITLE
Add support for pre-serialized events

### ISSUE THREAD

**Issue by nemerle** [Segs/Segs#122]
Events are translated into bitstreams on another thread than the one that constructed them, if they reference any object apart from themeselves, the state during construction and state during serialization might differ, and accessing that state might require costly thread synchronization. 

Consider adding PreserializedEvent
```cpp
class PreserializedEvent : MapLinkEvent 
{
    BitStream m_serialization_buffer;
    PreserializedEvent(MapLinkEvent &from)  : MapLinkEvent(MapEventTypes::evPreserialized)
    {
        from.serializeto(m_serialization_buffer);
    }
    void serializeto(BitStream &tgt) override { tgt.appendBitstream(m_serialization_buffer); }
};
```
as an added bonus, we could create specific events on the stack, and only ```new``` PreserializedEvents

**Comment by broxen** [comment]
Does this relate to your work in #514?

**Comment by nemerle** [comment]
This might be closed after the merge of #514 since the trouble described here should not occur with events that carry all of their information inside, but we still have to stay vigilant :D

**Comment by broxen** [comment]
Closing per last comment and #514
